common/src/View/ScaleObjectsToolController.cpp

```diff
@@ -0,0 +1,237 @@
+                return false;
+            
+            m_tool->updateDragFaces(inputState.pickResult());
+
+            if (m_tool->beginResize(inputState.pickResult())) {
+                m_tool->updateDragFaces(inputState.pickResult());
+                return true;
+            }
+            return false;
+        }
+        
+        bool ScaleObjectsToolController::doMouseDrag(const InputState& inputState) {
+            return updateResize(inputState);
+        }
+        
+        bool ScaleObjectsToolController::updateResize(const InputState& inputState) {
+            const bool vertical = inputState.modifierKeysDown(ModifierKeys::MKAlt);
+                        
+            return m_tool->resize(inputState.pickRay(),
+                                  inputState.camera(),
+                                  vertical);
+        }
+        
+        void ScaleObjectsToolController::doEndMouseDrag(const InputState& inputState) {
+            m_tool->commitResize();
+            m_tool->updateDragFaces(inputState.pickResult());
+        }
+        
+        void ScaleObjectsToolController::doCancelMouseDrag() {
+            m_tool->cancelResize();
+        }
+        
+        void ScaleObjectsToolController::doSetRenderOptions(const InputState& inputState, Renderer::RenderContext& renderContext) const {
+            
+            if (m_tool->isShearing()) {
+                renderContext.setForceHideSelectionGuide();
+            }
+            if (thisToolDragging() && !m_tool->isShearing()) {
+                renderContext.setForceShowSelectionGuide();
+            }
+            // TODO: force rendering of all other map views if the input applies and the tool has drag faces
+        }
+        
+        void ScaleObjectsToolController::doRender(const InputState& inputState, Renderer::RenderContext& renderContext, Renderer::RenderBatch& renderBatch) {
+            
+            
+            // regular indicators
+            
+            if (m_tool->isShearing()) { // && thisToolDragging()
+                Renderer::RenderService renderService(renderContext, renderBatch);
+                
+                // render sheared box
+                renderService.setForegroundColor(Color(0, 255, 0));
+                const auto mat = m_tool->bboxShearMatrix();
+                const auto op = [&](const Vec3& start, const Vec3& end){
+                    renderService.renderLine(mat * start, mat * end);
+                };
+                eachBBoxEdge(m_tool->bboxAtDragStart(), op);
+                
+                // render shear handle
+                
+                const Polygon3f poly = m_tool->shearHandle();
+                if (poly.vertexCount() != 0) {
+                    renderService.setForegroundColor(Color(128, 128, 255));
+                    renderService.renderPolygonOutline(poly.vertices());
+                }
+                return;
+            }
+
+            // bounds and corner handles
```

Comment:
This method is a bit long and it might be helpful to move it's parts into separate methods. As a rule of thumb, whenever I find a comment like this, it's a good indicator that the following code can be moved into a method of its own.